src/main.rs

```diff
@@ -4,27 +4,52 @@ use clap::Parser;
 use color_eyre::eyre::WrapErr;
 use interactive_clap::ToCliArgs;
 
-mod commands;
-mod common;
-mod config;
-mod js_command_match;
-mod network;
-mod network_for_transaction;
-mod network_view_at_block;
-mod transaction_signature_options;
-mod types;
-mod utils_command;
+pub use near_cli_rs::commands;
+pub use near_cli_rs::common;
+pub use near_cli_rs::config;
+pub use near_cli_rs::js_command_match;
+pub use near_cli_rs::network;
+pub use near_cli_rs::network_for_transaction;
+pub use near_cli_rs::network_view_at_block;
+pub use near_cli_rs::transaction_signature_options;
+pub use near_cli_rs::types;
+pub use near_cli_rs::utils_command;
 
 pub use common::CliResult;
-pub type GlobalContext = (crate::config::Config,);
+pub use near_cli_rs::GlobalContext;
 
 #[derive(Debug, Clone, interactive_clap::InteractiveClap)]
-#[interactive_clap(context = crate::GlobalContext)]
-struct Cmd {
+#[interactive_clap(input_context = crate::GlobalContext)]
+#[interactive_clap(output_context = CmdContext)]
```

Comment:
Instead of `GlobalContext` -> `CmdContext` and then later back to `GlobalContext` (through .into() implementation), make it the other way around: `input_context = ConfigContext, output_context = GlobalContext`, and `ConfigContext` would only hold `config` while `GlobalContext` will add `offline` to it.